src/nvim/screen.c

```diff
@@ -4199,8 +4199,22 @@ win_line (
             || (wp->w_p_list && lcs_eol != NUL && p_extra != at_end_str)
             || (n_extra != 0 && (c_extra != NUL || *p_extra != NUL)))
         ) {
+      // Special trick to make copy/paste of wrapped lines work with
+      // xterm/screen: write the next character after the end of the
+      // line. This will work with all terminal types (regardless of the
+      // xn,am settings).
+      bool wrap = wp->w_p_wrap     // Wrapping enabled.
+        && filler_todo <= 0        // Not drawing diff filler lines.
+        && lcs_eol_one != -1       // Haven't printed the lcs_eol character.
+        && row != endrow - 1       // Not the last line being displayed.
+        && wp->w_width == Columns  // Window spans the width of the screen.
+        && !wp->w_p_rl;            // Not right-to-left.
+      if (wrap) {
```

Comment:
I would place this after the `screen_line` call to make it immediately clear it will affect a following `screen_line` call, and not this one.